plenum/test/monitoring/test_request_measurement_class.py

```diff
@@ -0,0 +1,118 @@
+import pytest
+from plenum.server.monitor import RequestMeasurement
+
+
+ACCURACY = .1e-5
+
+
+@pytest.fixture(scope="function")
+def request_measurement():
+    rm = RequestMeasurement()
+    rm.first_ts = 1
+    rm.window_start_ts = 1
+    return rm
+
+
+def test_add_request(request_measurement):
+    assert request_measurement.add_request
+
+
+@pytest.mark.skip(reason="Not implemented yet")
+def test_get_avg_latency(request_measurement):
+    assert request_measurement.get_avg_latency
+
+
+@pytest.mark.skip(reason="Not implemented yet")
+def test_get_max_latency(request_measurement):
+    assert request_measurement.get_max_latency
+
+
+def test_get_throughput(request_measurement):
+    assert request_measurement.get_throughput
+
+
+@pytest.mark.skip(reason="Not implemented yet")
+def test_avg_latency(rm_with_random_requests, recv_ordered_ts):
+    assert abs(rm_with_random_requests.get_avg_latency() - sum([b - a for a, b in recv_ordered_ts]) / len(recv_ordered_ts)) < ACCURACY
+
+
+@pytest.mark.skip(reason="Not implemented yet")
+def test_max_latency(rm_with_random_requests, recv_ordered_ts):
+    assert rm_with_random_requests.get_max_latency() == max([b - a for a, b in recv_ordered_ts])
+
+
+def test_add_request_and_eval_first_throughput(request_measurement):
+    rm = request_measurement
+    ordered_ts = [x for x in range(1, rm.inner_window + 2)]
+    assert len(ordered_ts) == rm.inner_window + 1
+    for ts in ordered_ts:
+        request_measurement.add_request(ordered_ts=ts)
+    assert rm.throughput != 0
+    assert rm.window_start_ts == rm.first_ts + rm.inner_window
+
+
+def test_get_thoughput_ts_less_than_window(request_measurement):
+    rm = request_measurement
+    ordered_ts = [1, 2, 3, 5, rm.min_cnt * rm.inner_window]
+    for ts in ordered_ts:
+        rm.add_request(ordered_ts=ts)
+    assert rm.get_throughput(rm.min_cnt * rm.inner_window) is None
```

Comment:
This makes me think again that updating internals in `get_throughput` was not a good idea...